core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerServiceSpec.js

```diff
@@ -0,0 +1,344 @@
+    spyOn(statesToQuestionsService, 'getQuestions').andReturn(null);
+    expectedReturnedValue = simpleEditorManagerService.tryToInit();
+    expect(expectedReturnedValue).toBeFalsy();
+  });
+
+  it('should return data', function() {
+    // Expected initialized data inside simpleEditorManagerService
+    var expectedSimpleEditorManagerData = {
+      title: data.title,
+      introductionHtml: data.states.Introduction.content[0].value,
+      questionList: null
+    };
+    // Initialize simpleEditorManagerService.data
+    simpleEditorManagerService.tryToInit();
+    expectedSimpleEditorManagerData.questionList = questionList;
+    expect(simpleEditorManagerService.getData())
+      .toEqual(expectedSimpleEditorManagerData);
+  });
+
+  it('should return title', function() {
+    var expectedTitle = data.title;
+    // Initialize simpleEditorManagerService.data
+    simpleEditorManagerService.tryToInit();
+    expect(simpleEditorManagerService.getTitle()).toEqual(expectedTitle);
+  });
+
+  it('should return introduction html', function() {
+    var expectedIntroductionHtml = data.states.Introduction.content[0].value;
+    simpleEditorManagerService.tryToInit();
+    expect(simpleEditorManagerService.getIntroductionHtml())
+      .toEqual(expectedIntroductionHtml);
+  });
+
+  it('should return question list', function() {
+    var expectedSimpleEditorManagerQuestionList = questionList;
+    simpleEditorManagerService.tryToInit();
+    expect(simpleEditorManagerService.getQuestionList())
+      .toEqual(expectedSimpleEditorManagerQuestionList);
+  });
+
+  it('should save the given new title', function() {
+    var expectedNewTitle = 'ExpNewTitle';
+    simpleEditorManagerService.saveTitle(expectedNewTitle);
+    expect(explorationTitleService.savedMemento).toEqual(expectedNewTitle);
+    expect(simpleEditorManagerService.getTitle()).toEqual(expectedNewTitle);
+  });
+
+  it('should save the given new introduction html', function() {
+    var expectedNewIntroductionHtmll = '<p> new intro </p>';
+    simpleEditorManagerService
+      .saveIntroductionHtml(expectedNewIntroductionHtmll);
+    expect(simpleEditorShimService.getIntroductionHtml())
+      .toEqual(expectedNewIntroductionHtmll);
+    expect(simpleEditorManagerService.getIntroductionHtml())
+      .toEqual(expectedNewIntroductionHtmll);
+  });
+
+  it('should save the customization args', function() {
+    var testStateName = 'Introduction';
+    var expectedNewCustomizationArgs = {
+      choices: {
+        value: [
+          '<p>Option 3</p>'
+        ]
+      }
+    };
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.saveCustomizationArgs(
+      testStateName, expectedNewCustomizationArgs);
+    expect(explorationStatesService.getState(testStateName)
+      .interaction.customization_args).toEqual(expectedNewCustomizationArgs);
+  });
+  /**
+    * Not working, needs debugging.
+    It('should save the answer groups', function () {
```

Comment:
OK, I took a look. It doesn't seem too tricky to me to trace the error; did you look at the log? I see the following:

```
	TypeError: answerGroup.toBackendDict is not a function
	    at /core/templates/dev/head/pages/exploration_editor/EditorServices.js:9:30303
	    at Array.map (native)
	    at answer_groups (/core/templates/dev/head/pages/exploration_editor/EditorServices.js:9:30157)
	    at convertToBackendRepresentation (/core/templates/dev/head/pages/exploration_editor/EditorServices.js:9:33723)
	    at saveStateProperty (/core/templates/dev/head/pages/exploration_editor/EditorServices.js:9:32340)
	    at Object.saveInteractionAnswerGroups (/core/templates/dev/head/pages/exploration_editor/EditorServices.js:9:37371)
	    at Object.saveAnswerGroups (/core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorShimService.js:9:2450)
	    at Object.saveAnswerGroups (/core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerService.js:9:3511)
	    at .<anonymous> (/core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerServiceSpec.js:324:34)
```

which shows that the error happens due to the call to saveAnswerGroups with the JS list you've passed to it. Going through the relevant files I notice that this goes all the way to saveStateProperty and thence to convertToBackendRepresentation -- which means that the argument is expected to be an AnswerGroup domain object, not a JS dict. (See core/templates/dev/head/domain/exploration/AnswerGroupObjectFactory.js.)

If you take a look in the non-spec file (SimpleEditorManagerService.js) and look at how the relevant functions are called, I think you'll see that they're called with AnswerGroup domain objects as well.

So the fix is to change your test to use an AnswerGroup domain object instead of a JS dict. However, I hope the above also illustrates how to debug and analyze a failing test in general -- start at the stack trace, and track down the clues based on that. Let me know if you have questions about the above.

Thanks!